src/foam/dao/LRUCachingDAO.js

```diff
@@ -0,0 +1,154 @@
+/**
+ * @license
+ * Copyright 2018 The FOAM Authors. All Rights Reserved.
+ * http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+ foam.CLASS({
+  package: 'foam.dao',
+  name: 'LRUCachingDAO',
+  extends: 'foam.dao.ProxyDAO',
+  implements: [ 'foam.mlang.Expressions' ],
+
+  requires: [
+    'foam.dao.MDAO',
+    'foam.dao.QuickSink'
+  ],
+
+  documentation: `
+     Least Recently Used CachingDAO :
+     Manages a DAO\'s size by removing old items. Commonly applied inside a cache to limit the cache\'s size.
+  `,
+
+  classes: [
+    {
+      /** Links an object id to a last-accessed timestamp */
+      name: 'LRUCacheItem',
+      properties: [
+        {
+          name: 'id',
+        },
+        {
+          class: 'Int',
+          name: 'timestamp'
+        }
+      ]
+    }
+  ],
+
+  properties: [
+    {
+      /** The maximum size to allow the target dao to be. */
+      class: 'Int',
+      name: 'maxSize',
+      value: 100
+    },
+    {
+      /** Tracks the age of items in the target dao. */
+      name: 'trackingDAO',
+      of: 'foam.dao.DAO',
+      factory: function() {
+        return this.MDAO.create({ of: this.LRUCacheItem });
+      }
+    },
+    {
+      /** By starting at the current time */
+      class: 'Int',
+      name: 'lastTimeUsed_',
+      factory: function() { return Date.now(); }
+    },
+    {
+      class: 'foam.dao.DAOProperty',
+      name: 'dao'
+    }
+  ],
+
+  methods: [
+    function init() {
+      this.SUPER();
+      var self = this;
+
+      self.delegate.listen(self.QuickSink.create({
+        putFn: this.onPut,
+        removeFn: this.onRemove
+      }));
+    },
+
+    /** Calculates a timestamp to use in the tracking dao. */
+    function getTimestamp() {
+      // Just increment on each request.
+      return this.lastTimeUsed_++;
```

Comment:
If this is just a seqNo (which is good) then why make it be a timestamp at all, rather than just a seqNo starting at 0?